app/src/components/dialog/ErrorDialog.test.tsx

```diff
@@ -35,4 +35,38 @@ describe('ErrorDialog', () => {
 
     expect(baseElement).toMatchSnapshot();
   });
+
+  it('renders correctly with error details shown/hidden', async () => {
+    const { baseElement, getByText } = render(
+      <div id="root">
+        <ErrorDialog
+          dialogTitle="This is dialog title"
+          dialogText="This is dialog text"
+          dialogErrorDetails={['an error', { error: 'another error' }]}
+          dialogError="This is dialog error"
+          open={true}
+          onClose={() => jest.fn()}
+          onOk={() => jest.fn()}
+        />
+      </div>
+    );
+
+    fireEvent.click(getByText('Show detailed error message'));
+
+    await waitFor(() => {
+      expect(getByText('an error')).toBeVisible();
+      expect(getByText('{"error":"another error"}')).toBeVisible();
+    });
+
+    expect(baseElement).toMatchSnapshot();
+
+    fireEvent.click(getByText('Hide detailed error message'));
+
+    await waitFor(() => {
+      expect(getByText('an error')).not.toBeVisible();
+      expect(getByText('{"error":"another error"}')).not.toBeVisible();
+    });
+
+    expect(baseElement).toMatchSnapshot();
```

Comment:
I would remove this personally - I don't think this specific assertion or having an additional snapshot of this adds any value to the assertions you already have above which are testing everything we need to 💯 

let's remove this, keep the first `asSnapshot` assertion

Reply:
removed